discovery-provider/src/queries/get_health_test.py

```diff
@@ -0,0 +1,310 @@
+
+
+def test_get_health(web3_mock, redis_mock, db_mock):
+    """Tests that the health check returns db data"""
+    # Set up web3 eth
+    def getBlock(_u1, _u2):  # unused
+        block = MagicMock()
+        block.number = 2
+        block.hash = HexBytes(b"\x02")
+        return block
+    web3_mock.eth.getBlock = getBlock
+
+    # Set up db state
+    with db_mock.scoped_session() as session:
+        Block.__table__.create(db_mock._engine)
+        session.add(Block(
+            blockhash='0x01',
+            number=1,
+            parenthash='0x01',
+            is_current=True,
+        ))
+
+    args = {}
+    health_results, error = get_health(args)
+
+    assert error == False
+
+    assert health_results["web"]["blocknumber"] == 2
+    assert health_results["web"]["blockhash"] == "0x02"
+    assert health_results["db"]["number"] == 1
+    assert health_results["db"]["blockhash"] == "0x01"
+    assert health_results["block_difference"] == 1
+
+    assert "maximum_healthy_block_difference" in health_results
+    assert "version" in health_results
+    assert "service" in health_results
+
+
+def test_get_health_using_redis(web3_mock, redis_mock, db_mock):
+    """Tests that the health check returns redis data first"""
+    # Set up web3 eth
+    def getBlock(_u1, _u2):  # unused
+        block = MagicMock()
+        block.number = 2
+        block.hash = HexBytes(b"\x02")
+        return block
+    web3_mock.eth.getBlock = getBlock
+
+    # Set up redis state
+    redis_mock.set(latest_block_redis_key, '3')
+    redis_mock.set(latest_block_hash_redis_key, '0x3')
+    redis_mock.set(most_recent_indexed_block_redis_key, '2')
+    redis_mock.set(most_recent_indexed_block_hash_redis_key, '0x02')
+
+    # Set up db state
+    with db_mock.scoped_session() as session:
+        Block.__table__.create(db_mock._engine)
+        session.add(Block(
+            blockhash='0x01',
+            number=1,
+            parenthash='0x01',
+            is_current=True,
+        ))
+
+    args = {}
+    health_results, error = get_health(args)
+
+    assert error == False
+
+    assert health_results["web"]["blocknumber"] == 3
+    assert health_results["web"]["blockhash"] == "0x3"
+    assert health_results["db"]["number"] == 2
+    assert health_results["db"]["blockhash"] == "0x02"
+    assert health_results["block_difference"] == 1
+
+    assert "maximum_healthy_block_difference" in health_results
+    assert "version" in health_results
+    assert "service" in health_results
+
+
+def test_get_health_partial_redis(web3_mock, redis_mock, db_mock):
+    """Tests that the health check returns db data if redis data is only partial"""
+    # Set up web3 eth
+    def getBlock(_u1, _u2):  # unused
+        block = MagicMock()
+        block.number = 2
+        block.hash = HexBytes(b"\x02")
+        return block
+    web3_mock.eth.getBlock = getBlock
+
+    # Set up redis state
+    redis_mock.set(latest_block_redis_key, '3')
```

Comment:
this is so nice - even though it doesn't cover the actual population of the data, isolating behavior like this can confirm nothing breaks in our query layer